WordPress/openverse
# Use PDM to manage catalog dependencies (WIP)

## Fixes



Related to https://github.com/WordPress/openverse/issues/4166 by @dhruvkb 

## Description


This is just an experiment, and I'm pretty sure it doesn't work. While Airflow and such all _run_ (I can get to the UI), there's _something_ wonky going on with multithreading. I cannot run tests, and when I try to run any DAGs, they get weirdly stuck, _I think_, or else they are moving very slowly. Tests just hang, but I can run some individual tests (I wasn't able to find a specific individual test that got stuck, but I ran out of time to keep trying to find one).

Maybe this is a fool's errand though, and we should just accept the continued use of pip for the catalog. But airflow itself uses hatch, so why should we be stuck on requirements.txt and such?

Well, I just need to look into this more. I just picked this up as a curiosity becuase I am stuck on other work I have going on.

## Testing Instructions

Try running Airflow locally and the tests. Maybe CI will pass (but I'm logging off right after pushing this, so I won't see until tomorrow).

## Checklist


      PRs) or the media properties generator (`just catalog/generate-docs media-props`
      for the catalog or `just api/generate-docs` for the API) where applicable.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
